mycroft/client/wifisetup/main.py

```diff
@@ -218,12 +402,20 @@ def connect(self, event=None):
                 connected = self.get_connected(ssid)
                 if connected:
                     wpa(self.iface, 'save_config')
-                    # ConfigurationManager.set(self.NAME, 'setup', False, True)
 
             self.client.emit(Message("mycroft.wifi.connected",
                                      {'connected': connected}))
             LOG.info("Connection status for %s = %s" % (ssid, connected))
 
+            if connected:
```

Comment:
The state of 'connected' changes in the body of the else, this code can't be moved to the top check.  This catches when you are connected after both the "then" and the "else".